### DISCUSSION TITLE
Let Superset provide another canvas based presentation/让Superset提供另一种基于画布方式的展现形式

### DISCUSSION THREAD

**Discussion by cdmikechen** [apache/superset#18526] [Ideas]
**Is your feature request related to a problem? Please describe.**

`Superset`是个非常好的数据可视化产品，我和我的公司是在 0.35 版本开始接触它的。我们使用它在接下来的时间里为客户提供了很多很有用的服务。
`Superset ` is a very good data visualization product. My company and me began to use it from 0.35. We used it to provide many useful services for customers in the next time.
原来我们的基于 `Superset` 的工作主要在如何适配本土的中文化上面。在今年年初的时候，我们在结合以往的使用经验和一些需求，决定在基于图表的基础上，脱离现有看板模式的约束，让整个展现页面进行拖拽化的功能升级。对于新的功能，我们暂且称之为 `大屏`。
Originally, our work based on `Superset` mainly focused on how to adapt to Chinese. At the beginning of this year, we made a decision after combining our previous experience and some needs. We are ready to use the chart based function, break away from the constraints of the existing dashboard mode, and upgrade the drag and drop function of the whole presentation page. For the new function, we call it `big screen` for the time being.

**Describe the solution you'd like**

新功能的设计理念主要是借鉴了国内阿里的数据可视化产品 [DataV](https://www.aliyun.com/product/bigdata/datav)，主要的理念还是将整个页面上的图表不规约于已有的行列线内，而是可以自由进行布局。同时辅以一些定制的背景图片，主题色系，贴图等等，让整个界面的画面更加的生动。画布上的图表还是会沿用 `superset-ui` 内的组件，在框架的功能和风格上仍保持和 `Superset` 的一致性。
The design concept of the new function is mainly based on Alibaba's data visualization product [DataV]( https://www.aliyun.com/product/bigdata/datav). The main idea is that the charts on the whole page are not limited to the existing row and column lines, but can be arranged freely. At the same time, it is supplemented by some customized background pictures, theme color system, maps and so on to make the picture of the whole interface more vivid. The chart on the canvas will still use the components in the `superset-ui`, and the function and style of the framework will remain consistent with that of the `Superset`.

**Describe alternatives you've considered**

针对整个设计，我们已经在内部花了几个月的时间基于 `1.0.1` 版本（在当时最新的版本）做了一个demo出来。整个demo基本上将上述的功能都进行了实现，具体的截图会放在后面。我们觉得虽然这个独立的需求和看板是两种不同的设计思路，但是对于拓宽 `Superset` 的使用场景会是一个很好的帮助。具体的代码我们也会在未来的时间开放出来并在合适的实际看是否已PR形式提交到 `Superset` 主分支内。
For the whole design, we have spent several months internally to make a demo based on the `1.0.1` (the latest version at that time). The above functions are basically implemented in the whole demo, and the specific screenshots will be put later. We think that although this independent requirement and `Dashboard` are two different design ideas, it will be a good help to broaden the use scenario of `Superset` We will also open the codes in the future and see if it has been submitted to the `Superset` main branch by some PRs.

**Additional context**

以下的图片都是基于该demo在 `Superset` 上做的实际的页面，而不是设计图。注意这个颜色风格我们自己调整过。
The following pictures are based on the actual page made by the demo on `Superset`, not the design drawing. Note that this color style has been adjusted by ourselves.

![Xnip2021-11-04_22-30-57](https://user-images.githubusercontent.com/12069428/140333825-00ee9895-22e9-4bbb-b825-7a442bd9d6d7.jpg)

![Xnip2021-11-04_22-31-51](https://user-images.githubusercontent.com/12069428/140333950-bdf1953b-8ab6-42bf-aa12-a99cb626621e.jpg)

**Comment by junlincc** [comment]
OMG. thanks for sharing! @cdmikechen this looks awesome. Please reach out to me in Slack - junlin chen. we would like to take a look at your fork and hopefully present to the wider community!! 

cc @zhaoyongjie @stephenLYZ

**Comment by chuancyzhang** [comment]
looks cool

**Comment by aniaan** [comment]
cool
